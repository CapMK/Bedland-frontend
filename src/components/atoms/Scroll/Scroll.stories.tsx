import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import Scroll from './Scroll';

const meta = {
  title: 'Atoms/Scroll',
  component: Scroll,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof Scroll>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: (
      <div
        style={{
          border: '1px solid black',
          height: '12.5rem',
          width: '18.75rem',
          color: 'white',
        }}
      >
        Test
      </div>
    ),
    contentSize: 5,
  },
};
