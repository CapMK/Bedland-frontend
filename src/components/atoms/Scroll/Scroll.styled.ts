import styled from 'styled-components';
import { PaletteType } from 'theme/theme.types';

const THUMB_WIDTH = 0.5;
const SCROLLBAR_RIGHT_PADDING = 0.375;
const SCROLLBAR_BORDER_RADIUS = 8;
const SCROLLBAR_WIDTH = THUMB_WIDTH + 2 * SCROLLBAR_RIGHT_PADDING;

export const ScrollBar = styled.div<{
  contentSize: number;
  palette: PaletteType;
}>`
  height: ${({ contentSize }) => contentSize}rem;
  overflow-y: scroll;

  @supports (overflow: overlay) {
    overflow: overlay;
  }

  scrollbar-width: thin;

  ::-webkit-scrollbar {
    width: ${SCROLLBAR_WIDTH}rem;
  }

  ::-webkit-scrollbar-thumb {
    background: ${({ palette }) => palette.blueDark};
    border-radius: ${2 * SCROLLBAR_BORDER_RADIUS}px;
    border: ${SCROLLBAR_RIGHT_PADDING}rem solid transparent;
    background-clip: padding-box;
  }

  ::-webkit-scrollbar-track {
    margin-bottom: 0.5rem;
    scrollbar-color: ${({ palette }) => palette.white};
  }
`;

export default ScrollBar;
