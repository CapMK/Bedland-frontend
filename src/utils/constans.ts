export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
