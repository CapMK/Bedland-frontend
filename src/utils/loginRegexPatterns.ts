import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
