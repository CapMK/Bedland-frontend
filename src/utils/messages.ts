const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
