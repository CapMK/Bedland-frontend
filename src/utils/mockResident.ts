const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
