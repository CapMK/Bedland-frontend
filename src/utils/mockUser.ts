import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
