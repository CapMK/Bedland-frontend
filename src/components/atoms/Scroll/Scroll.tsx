import { PropsWithChildren, useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import * as Styled from './Scroll.styled';

export type ScrollProps = {
  contentSize: number;
};

const Scroll = ({ children, contentSize }: PropsWithChildren<ScrollProps>) => {
  const { palette } = useContext(ThemeContext);
  return (
    <Styled.ScrollBar contentSize={contentSize} palette={palette}>
      {children}
    </Styled.ScrollBar>
  );
};

export default Scroll;
