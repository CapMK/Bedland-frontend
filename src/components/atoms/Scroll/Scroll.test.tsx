import { renderWithProviders } from 'utils/test.utils';
import Scroll, { ScrollProps } from './Scroll';

describe('Scroll', () => {
  const item = ({ contentSize }: ScrollProps) =>
    renderWithProviders(<Scroll contentSize={contentSize} />, {});
  it('should renders correctly Scroll', () => {
    expect(
      item({
        contentSize: 2,
      }),
    ).toMatchSnapshot();
  });
});
