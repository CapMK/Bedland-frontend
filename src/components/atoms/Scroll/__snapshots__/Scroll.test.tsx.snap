// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Scroll > should renders correctly Scroll 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  height: 2rem;
  overflow-y: scroll;
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
}

.c0::-webkit-scrollbar {
  width: 1.25rem;
}

.c0::-webkit-scrollbar-thumb {
  background: #4553DD;
  border-radius: 16px;
  border: 0.375rem solid transparent;
  background-clip: padding-box;
}

.c0::-webkit-scrollbar-track {
  margin-bottom: 0.5rem;
  -webkit-scrollbar-color: #fff;
  -moz-scrollbar-color: #fff;
  -ms-scrollbar-color: #fff;
  scrollbar-color: #fff;
}

@supports (overflow:overlay) {
  .c0 {
    overflow: overlay;
  }
}

<body>
    <div>
      <div
        class="c0"
      />
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  height: 2rem;
  overflow-y: scroll;
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
}

.c0::-webkit-scrollbar {
  width: 1.25rem;
}

.c0::-webkit-scrollbar-thumb {
  background: #4553DD;
  border-radius: 16px;
  border: 0.375rem solid transparent;
  background-clip: padding-box;
}

.c0::-webkit-scrollbar-track {
  margin-bottom: 0.5rem;
  -webkit-scrollbar-color: #fff;
  -moz-scrollbar-color: #fff;
  -ms-scrollbar-color: #fff;
  scrollbar-color: #fff;
}

@supports (overflow:overlay) {
  .c0 {
    overflow: overlay;
  }
}

<div>
    <div
      class="c0"
    />
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
